backend/code_review_backend/issues/serializers.py

```diff
@@ -26,20 +28,48 @@ class RepositorySerializer(serializers.ModelSerializer):
 
     class Meta:
         model = Repository
-        fields = ("id", "phid", "slug", "url")
+        fields = ("id", "slug", "url")
+
+
+class RepositoryGetOrCreateField(serializers.SlugRelatedField):
+    help_text = (
+        "Get or create a repository. URL must match allowed hosts from settings."
+    )
+    default_error_messages = {
+        **serializers.SlugRelatedField.default_error_messages,
+        "invalid_url": "Repository URL must match hg.mozilla.org.",
+    }
+    queryset = Repository.objects.all()
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs, slug_field="url")
+
+    def to_internal_value(self, url):
+        parsed = urlparse(url)
+        if parsed.netloc not in settings.ALLOWED_REPOSITORY_HOSTS:
+            try:
+                return self.get_queryset().get(url=url)
+            except Repository.DoesNotExist:
+                self.fail("invalid_url")
+            except (TypeError, ValueError):
+                self.fail("invalid")
+                return
+        try:
+            repo, _ = self.get_queryset().get_or_create(
+                url=url, defaults={"slug": parsed.path}
```

Comment:
Please remove the leading `/` from `parsed.path` so we get `mozilla-central` instead of `/mozilla-central`